aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java

```diff
@@ -348,79 +413,230 @@ private void syncHostTemperatureAndHumidySensor(boolean fullSync) {
-          */
-         String frontPanelSensor = null;
-         String backPanelSensor = null;
-         List<String> tempSensors = new ArrayList<String>();
-         List<String> humiditySensors = new ArrayList<String>();
-         boolean needUpdate=false;
-         for (String pduID : pdus) {
-            if (temperatureSensorMap.containsKey(pduID)) {
-               tempSensors.addAll(temperatureSensorMap.get(pduID));
-            }
-            if(humiditySensorMap.containsKey(pduID)) {
-               humiditySensors.addAll(humiditySensorMap.get(pduID));
-            }
+         Map<String, String> temperatureSensorAssetIds = new HashMap<String, String>();
+         Map<String, String> humiditySensorAssetIds = new HashMap<String, String>();
+         boolean needUpdate = false;
+         Map<String, Map<String,Map<String,String>>> metricsFormular = server.getMetricsformulars();
+         Map<String,Map<String,String>> sensorMetricsNameAndIdMap = metricsFormular.get(FlowgateConstant.SENSOR);
+         if(sensorMetricsNameAndIdMap == null || sensorMetricsNameAndIdMap.isEmpty()) {
+            sensorMetricsNameAndIdMap = new HashMap<String,Map<String,String>>();
          }
-         if (!tempSensors.isEmpty()) {
-            frontPanelSensor = backPanelSensor = tempSensors.get(0);
-            if (tempSensors.size() > 1) {
-               backPanelSensor = tempSensors.get(1);
-            }
-            if (!server.getSensorsformulars().containsKey(ServerSensorType.FRONTPANELTEMP)) {
-               server.getSensorsformulars().put(ServerSensorType.FRONTPANELTEMP, frontPanelSensor);
-               needUpdate = true;
+         for (String pduID : pdus) {
+            Map<String,Map<String,String>> metricNameAndSensorIdMap = sensorMap.get(pduID);
+            if(metricNameAndSensorIdMap == null) {
+               continue;
             }
-            if (!server.getSensorsformulars().containsKey(ServerSensorType.BACKPANELTEMP)) {
-               server.getSensorsformulars().put(ServerSensorType.BACKPANELTEMP, backPanelSensor);
-               needUpdate = true;
+            for(Map.Entry<String, Map<String,String>> map : metricNameAndSensorIdMap.entrySet()) {
+               switch (map.getKey()) {
+               case MetricName.SERVER_FRONT_TEMPERATURE:
+                  Map<String,String> oldFrontTempdMap = sensorMetricsNameAndIdMap.get(MetricName.SERVER_FRONT_TEMPERATURE);
+                  if(oldFrontTempdMap == null) {
+                     sensorMetricsNameAndIdMap.put(MetricName.SERVER_FRONT_TEMPERATURE, map.getValue());
+                     needUpdate = true;
+                  }else {
+                     /**
+                      * Suggest use cabinetAssetNumber+cabinet_unit_position as key,
+                      * because of a server may have many pdus, every pdu has itself cabinet,
+                      * the cabinet_unit_position between every cabinet may be duplicated,
```

Comment:
this comment is not correct. usually server will not connect to pdu located on other cabinet.

Reply:
Removed